xilem_core/src/view.rs

```diff
@@ -9,6 +9,10 @@ use alloc::{boxed::Box, sync::Arc};
 
 use crate::{message::MessageResult, DynMessage, Mut, ViewElement};
 
+/// This trait is a necessary workaround of the orphan rules to be able to have `impl ViewSequence for impl View` to avoid ambiguity.
+/// It needs to be implemented for every type that implements [`View`].
+pub trait ViewMarker {}
```

Comment:
This probably needs much better docs, and needs to be mentioned in the docs for `View`. 

This will easily become the most confusing part of Xilem, especially if we don't explain this properly.

Reply:
> This will easily become the most confusing part of Xilem, especially if we don't explain this properly.

I don't think so, unless someone really wants to know why this is necessary, otherwise it's just (for me at least) "ok I also need to implement that type for the view", but yeah the drive/curiosity to understand *why* this is necessary may be there... 

Do you have suggestions for the documentation (or some direction at least)? I'm not sure how much we need to go into detail here.

The error messages, when removing that requirement in the `ViewSequence` impl also doesn't make sense:

```
conflicting implementation for `Option<_>`
    |
    = note: downstream crates may implement trait `view::View<_, _, _, _>` for type `core::option::Option<_>`
```

How should downstream crates implement that trait anyway... argh, confusing edge cases of rusts type-system...

-------

Lol, I just tried removing `ViewMarker` in the `ViewSequence` impl with `RUSTFLAGS="-Znext-solver"` and it works :) (at least in xilem_web, xilem currently doesn't compile, because a dependent crate (`async-io`) fails type-checking)

So maybe we should just put some warning, that this is a workaround of the current rust compiler, but that this limitation will be lifted in the future trait solver, to avoid people loosing their sanity trying to understand this? (I hope at least, that the new trait-solver doesn't implement that guard) Maybe really something to ask the types-team of Rust about.

Reply:
How does this sound:

Because [`View`] is generic, Rust [allows you](https://doc.rust-lang.org/reference/items/implementations.html#orphan-rules) to implement this trait for certain non-local types.
These non-local types can include `Vec<_>` and `Option<_>`.
If this trait were not present, those implementations of `View` would conflict with those types' implementations of `ViewSequence`.
This is because every `View` type also implementations `ViewSequence`.
Since `ViewMarker` is not generic, these non-local implementations are not permitted for this trait, which means that the conflicting implementation cannot happen.

Reply:
Yeah that makes sense, I'll add that. I just checked, removing `ViewMarker` from the super-bound of `View` produces that error again :/ but makes sense thinking about it...

Reply:
I've added that for the doc-comment of `ViewMarker`, should the doc-comment for `View` be more detailed?